packages/app/src/providers/h5wasm/models.ts

```diff
@@ -0,0 +1,44 @@
+import type { AttributeValues, EntityKind } from '@h5web/shared';
+
+export interface H5WasmEntityResponse {
+  name: string;
+  type:
+    | EntityKind.Dataset
+    | EntityKind.Group
+    | 'external_link'
+    | 'soft_link'
+    | 'other';
```

Comment:
At the moment soft links are directly resolved and not returned as "links" from h5wasm.  External links don't mean much in the context of a URL-based file, since URLs are not supported by HDF5 External Links, and they are not handled by h5wasm.
If it would help with compatibility I could certainly add special handling for these - though I don't see the benefit of identifying soft links instead of just resolving them.

Reply:
In H5Grove, soft links are also resolved by default, so the front-end doesn't know they are soft links. We return a `soft_link` entity only when H5Grove encounters a broken soft link that it cannot resolve. Do you know how h5wasm behave in such case?

Reply:
I'm sure it just breaks!  I can add some handling for that case.

Reply:
No rush, it can break for now, and we can keep the soft links out of the code until it doesn't, no worries.

Reply:
file.get(broken_soft_link) currently returns "undefined" from h5wasm.

Reply:
I just updated h5wasm so file.get(path) returns an object 
```js
{ type: "BrokenSoftLink", target: "/not/an/existing/object" }
``` 
for dangling soft links, and this for external links:
```js
{ type: "ExternalLink", filename: "external_filename", obj_path: "/path/in/external/file" }
```
(it is not yet published to npm)